controllers/nstemplateset/nstemplatetier.go

```diff
@@ -76,8 +78,15 @@ type tierTemplate struct {
 // process processes the template inside of the tierTemplate object and replaces the USERNAME variable with the given username.
 // Optionally, it also filters the result to return a subset of the template objects.
 func (t *tierTemplate) process(scheme *runtime.Scheme, username string, filters ...template.FilterFunc) ([]runtimeclient.Object, error) {
+	ns, err := configuration.GetWatchNamespace()
+	if err != nil {
+		return nil, err
+	}
```

Comment:
This could be moved to `getTierTemplate` function and store the namespace as part of `tierTemplate`
https://github.com/codeready-toolchain/member-operator/blob/master/controllers/nstemplateset/nstemplatetier.go#L24-L44
but it's not a big deal - your version is fine & valid too.